BlueprintUI/Sources/Internal/ElementIdentifier.swift

```diff
@@ -41,42 +41,82 @@
 
  You will note that the identifiers remain stable, which ultimately ensures that views are reused.
  */
-struct ElementIdentifier: Hashable, CustomDebugStringConvertible {
+final class ElementIdentifier: Hashable, CustomDebugStringConvertible {
 
-    let elementType: ObjectIdentifier
+    let elementType: Element.Type
+    let elementTypeIdentifier: ObjectIdentifier
     let key: AnyHashable?
 
     let count: Int
 
-    init(elementType: Element.Type, key: AnyHashable?, count: Int) {
+    private let hash: Int
+    private static var cachedIdentifiers: [ObjectIdentifier: [Int: ElementIdentifier]] = [:]
 
-        self.elementType = ObjectIdentifier(elementType)
+    static func identifier(for element: Element, key: AnyHashable?, count: Int) -> ElementIdentifier {
+
+        let elementType = type(of: element)
+
+        guard key == nil else {
+            let result = ElementIdentifier(elementType: elementType, key: key, count: count)
```

Comment:
(My fault again sorry) can just be: `return ElementIdentifier(elementType: elementType, key: key, count: count)`